README
wxMacMolPlt is a wxWidgets rewrite of MacMolPlt by Brett Bode's research group at
Iowa State University. The source code to this program is released under the 
terms of the license file in this directory.

Currently wxMacMolPlt includes build files and has been tested on Mac OS X,
Linux and Windows. Since most people building from source will be running
Linux this file focuses on the build instructions for Linux. To build from
source on any OS you must have a C++ compiler such as g++ and an install
of wxWidgets with OpenGL support enabled. wxWidgets 2.6 and later have been
used in the past, but current testing is on 2.8 and later.
Prebuilt packages of wx should also work and are generally preferrable.
Newer versions such as 2.8.x and 3.0.x have been tested on many recent 
Linux distros and are preferred.

For further build instructions refer the INSTALL file in this directory.

An html manual is included that includes a quick walkthrough of the program's
capabilities.



debian/changelog
wxmacmolplt (7.7.3-1) Bullseye; urgency=low

  * New upstream release

 -- Brett Bode <MMP.Guy@gmail.com>  Fri, 29 Dec 2023 12:01:27 -0500

wxmacmolplt (7.7.2-1) Bullseye; urgency=low

  * New upstream release

 -- Brett Bode <MMP.Guy@gmail.com>  Sun, 26 Sep 2021 12:01:27 -0500

wxmacmolplt (7.7.1-1) Jessie; urgency=low

  * New upstream version

 -- Brett Bode <MMP.Guy@gmail.com>  Tue, 03 Aug 2021 15:24:40 -0500

wxmacmolplt (7.7-1) precise; urgency=low

  * New upstream version

 -- Brett Bode <MMP.Guy@gmail.com>  Thu, 11 Aug 2016 17:14:02 -0500

wxmacmolplt (7.6.2-1) Jessie; urgency=low

  * New upstream version

 -- Brett Bode <MMP.Guy@gmail.com> Wed, 03 Feb 2016 21:27:57 -0600

wxmacmolplt (7.6.1-1) precise; urgency=low

  * New upstream version

 -- Brett Bode <MMP.Guy@gmail.com> Sat, 15 Aug 2015 15:38:54 -0500

wxmacmolplt (7.6-1) precise; urgency=low

  * New upstream version

 -- Brett Bode <MMP.Guy@gmail.com> Sat, 02 May 2015 21:18:35 -0500

wxmacmolplt (7.5-1) precise; urgency=low

  * New upstream version

 -- Brett Bode <brett@scl.ameslab.gov> Mon, 22 Jul 2013 09:18:13 -0500

wxmacmolplt (7.4.4-1) unstable; urgency=low

  * New upstream version

 -- root <root@debian.iprt.iastate.edu> Wed, 8 Aug 2012 18:22:09 -0500

wxmacmolplt (7.4.3-1) unstable; urgency=low

  * New upstream version

 -- brett <brett@scl.ameslab.gov> Mon, 30 Jan 2012 19:18:08 -0600

wxmacmolplt (7.4.2-1) unstable; urgency=low

  * New upstream version

 -- root <root@debian.iprt.iastate.edu> Sat, 04 Dec 2010 14:51:44 -0600

wxmacmolplt (7.4.1-1) unstable; urgency=low

  * New upstream version

 -- root <root@debian.iprt.iastate.edu> Tue, 13 Jul 2010 21:44:54 -0500

wxmacmolplt (7.4-1) unstable; urgency=low

  * New upstream version

 -- root <root@debian.iprt.iastate.edu> Wed, 11 Nov 2009 16:12:22 -0600

wxmacmolplt (7.3) unstable; urgency=low

  * New upstream version.

 -- root <brett@scl.ameslag.gov> Mon, 8 Dec 2008 20:57:30 -0600

wxmacmolplt (7.2.1-1) unstable; urgency=low

  * New bug fix upstream release.

 -- root <root@debian.iprt.iastate.edu> Tue, 13 May 2008 06:41:01 -0500

wxmacmolplt (7.2.0-1) unstable; urgency=low

  * New major upstream release.

 -- Brett Bode <brett@scl.ameslab.gov> Tue, 15 Apr 2008 09:39:15 -0500

wxmacmolplt (7.1.0-1) unstable; urgency=low

  * New major upstream release.

 -- Brett Bode <brett@scl.ameslab.gov> Thu, 13 Dec 2007 14:50:29 -0600

wxmacmolplt (7.0.0-2) unstable; urgency=low

  * Fix a build dependancy problem to get flash export working.

 -- Brett Bode <brett@scl.ameslab.gov> Fri, 17 Aug 2007 12:32:33 -0500

wxmacmolplt (7.0.0-1) unstable; urgency=low

  * New major upstream release
  * -

 -- Brett Bode <brett@scl.ameslab.gov> Fri, 17 Aug 2007 12:31:57 -0500

wxmacmolplt (6.5-1) unstable; urgency=low

  * New upstream release

 -- Brett Bode <brett@scl.ameslab.gov> Fri, 18 May 2007 10:12:34 -0500

wxmacmolplt (6.4-1) unstable; urgency=low

  * New upstream release

 -- Brett Bode <brett@scl.ameslab.gov> Wed, 21 Mar 2007 12:46:12 -0500

wxmacmolplt (6.3.1-1) unstable; urgency=low

  * New upstream release

 -- Brett Bode <brett@scl.ameslab.gov> Thu, 15 Feb 2007 20:58:19 -0600

wxmacmolplt (6.3.0-1) unstable; urgency=low

  * New upstream release

 -- Brett Bode <brett@scl.ameslab.gov> Wed, 7 Feb 2007 11:37:19 -0600

wxmacmolplt (6.2.0-1) unstable; urgency=low

  * Initial release

 -- Brett Bode <brett@scl.ameslab.gov> Tue, 10 Oct 2006 13:14:32 -0500



src/glf.h
/*
==============================================================================
|  GLF Library
|  Version 1.4
| 
|  Author: Roman Podobedov
|  Email: romka@ut.ee
|  WEB: http://romka.demonews.com
|  Release Date: 18 May 2001
|  
|  Copyright (C) 2000-2001, Romka Graphics
|  This library is freely distributable without any license or permissions
|  for non-commercial usage. You can use this library in any non-commercial
|  program. In each program, where You use this library You should keep 
|  this header (author name and coordinates)!
|  For commercial usage, please contact me: romka@ut.ee
==============================================================================
*/

#ifndef GLF_H
#define GLF_H

#include "mmp_gl.h"

#ifdef __cplusplus
extern "C" {
#endif

/* CONSTANTS */

#define GLF_ERROR           -1
#define GLF_OK               0
#define GLF_YES              1
#define GLF_NO               2

#define GLF_CONSOLE_MESSAGES 10
#define GLF_TEXTURING        11
#define GLF_CONTOURING       12

#define GLF_LEFT_UP          20
#define GLF_LEFT_CENTER      21
#define GLF_LEFT_DOWN        22
#define GLF_CENTER_UP        23
#define GLF_CENTER_CENTER    24
#define GLF_CENTER_DOWN      25
#define GLF_RIGHT_UP         26
#define GLF_RIGHT_CENTER     27
#define GLF_RIGHT_DOWN       28
#define GLF_CENTER           24

#define GLF_LEFT			 1
#define GLF_RIGHT			 2
#define GLF_UP				 3
#define GLF_DOWN			 4

/* Console constants */
#define GLF_CONSOLE_CURSOR   30

/* Library initialization (must be called before any usage of library) */
void glfInit(); 
/* Library closing (must be called after usage of library) */
void glfClose();

/* Font loading/unloading functions */
int	glfLoadFont(const char *font_name);			/* Load Vector font to memory */
int	glfLoadBMFFont(const char *FName);			/* Load Bitmap font to memory */
int	glfUnloadFont();						/* Unload font from memory */
int glfUnloadBMFFont();						/* Unload current BMF font */
int	glfUnloadFontD(int font_descriptor);	/* Unload font by font_descriptor */
int	glfUnloadBMFFontD(int bmf_descriptor);	/* Unload BMF font by bmf_descriptor */

/* Text drawing functions */
/* --------- Vector Fonts ---------------- */
void glfDrawWiredSymbol(char s);    /* Draw wired symbol    */
void glfDrawWiredString(char *s);   /* Draw wired string    */
void glfDrawSolidSymbol(char s);    /* Draw solid symbol    */
void glfDrawSolidString(const char *s);   /* Draw wired string    */
void glfDraw3DWiredSymbol(char s);  /* Draw 3D wired symbol */
void glfDraw3DWiredString(char *s); /* Draw 3D wired string */
void glfDraw3DSolidSymbol(char s);  /* Draw 3D solid symbol */
void glfDraw3DSolidString(char *s); /* Draw 3D solid string */

/* --------- Bitmap Fonts ---------------- */
void glfStartBitmapDrawing();       /* Must be called before bitmap text drawing */
void glfStopBitmapDrawing();        /* Must be called after bitmap text drawing */
void glfDrawBSymbol(char s);        /* Draw one symbol by bitmapped font */
void glfDrawBString(const char *s);       /* Draw string by bitmapped font */
void glfDrawBMaskSymbol(char s);	/* Draw one symbol by bitmapped font (using mask) */
void glfDrawBMaskString(const char *s);	/* Draw string by bitmapped font (using mask) */


/* Text drawing functions by specified font */
void glfDrawWiredSymbolF(int font_descriptor, char s);
void glfDrawWiredStringF(int font_descriptor, char *s);
void glfDrawSolidSymbolF(int font_descriptor, char s);
void glfDrawSolidStringF(int font_descriptor, char *s);
void glfDraw3DWiredSymbolF(int font_descriptor, char s);
void glfDraw3DWiredStringF(int font_descriptor, char *s);
void glfDraw3DSolidSymbolF(int font_descriptor, char s);
void glfDraw3DSolidStringF(int font_descriptor, char *s);

/* Text control functions */
void glfGetStringBoundsF(int fd, char *s, float *minx, float *miny, float *maxx, float *maxy);
void glfGetStringBounds(char *s, float *minx, float *miny, float *maxx, float *maxy);

void  glfSetSymbolSpace(float sp);   /* Set space between symbols */
float glfGetSymbolSpace();           /* Get space between symbols */
void  glfSetSpaceSize(float sp);     /* Set space size            */
float glfGetSpaceSize();             /* Get current space size    */

void  glfSetSymbolDepth(float dpth); /* Set depth of 3D symbol */
float glfGetSymbolDepth();           /* Get depth of 3D symbol */

int   glfSetCurrentFont(int Font_Descriptor);		/* Set current font */
int   glfSetCurrentBMFFont(int Font_Descriptor);	/* Set current bitmap font */
int   glfGetCurrentFont();							/* Get current font descriptor */
int   glfGetCurrentBMFFont();		                /* Get current bitmap font descriptor */

void  glfSetAnchorPoint(int anchp); /* Set symbol anchor point */
void  glfSetContourColor(float r, float g, float b, float a); /* Contour color */

/* Enable or Disable GLF features */

void glfEnable(int what); /* Enable GLF feature 'what' */
void glfDisable(int what); /* Disable GLF feature 'what' */

/* Console functions */
void glfSetConsoleParam(int width, int height);
int  glfSetConsoleFont(int Font_Descriptor);
void glfConsoleClear();
void glfPrint(char *s, int lenght);
void glfPrintString(char *s);
void glfPrintChar(char s);

void glfConsoleDraw();
void glfSetCursorBlinkRate(int Rate);

/* -------- Centering and direction ------------ */
void glfStringCentering(GLboolean center); /* Set string centering for vector fonts */
GLboolean glfGetStringCentering(); /* Get string centering for vector fonts */

void glfBitmapStringCentering(GLboolean center);  /* Set string centering for bitmap fonts */
GLboolean glfBitmapGetStringCentering();  /* Set string centering for bitmap fonts */

/* String direction for vector font (GLF_LEFT, GLF_RIGHT, GLF_UP, GLF_DOWN) */
/* GLF_LEFT by default */
void glfStringDirection(GLuint direction);
GLuint glfGetStringDirection();

/* -------- Rotating ------------ */
void glfSetRotateAngle(float angle); /* Set rotate angle for vector fonts */
void glfSetBRotateAngle(float angle); /* Set rotate angle for bitmap fonts */



#ifdef __cplusplus
};
#endif

#endif
